Modules/Shop/Http/Services/ShopService.php

```diff
@@ -371,23 +368,23 @@ public function getProductFilters($request): array
                 }
                 break;
         }
-
+        echo  $allProductQuery;die;
         $products = $allProductQuery->get();
         if ($products) {
-            foreach ($products as $v) {
+            foreach ($products as $productDetails) {
 
-                if (!array_key_exists($v->country, $brands)) {
-                    $countryDetails = Country::find($v->country);
-                    $countries[$v->country] = array(
-                        "id" => $v->country,
+                if (!array_key_exists($productDetails->country, $brands)) {
+                    $countryDetails = Country::find($productDetails->country);
+                    $countries[$productDetails->country] = array(
+                        "id" => $productDetails->country,
                         "value" => $countryDetails->name
                     );
                 }
```

Comment:
Instead of getting the country directly from the DB insdie the loop, get countries name of all the products, make them unique and then find all the countries from DB using whereIn. Then insdie the loop, get the specific country for the product form the collection of countries. Like this
$countryIds = collect($products)->pluck('country')->unique()->toArray();

$countries = Country::whereIn('id', $countryIds)
    ->pluck('name', 'id')
    ->toArray();

foreach ($products as $productDetails) {
    if (!array_key_exists($productDetails->country, $countries)) {
        $countries[$productDetails->country] = [
            'id' => $productDetails->country,
            'value' => $countries[$productDetails->country] ?? null
        ];
    }
}